dashboard/app/models/user.rb

```diff
@@ -2722,6 +2722,20 @@ def personal_account?
     end
   end
 
+  # Determine a student' state based on their teacher's state. Used for student
+  # accounts created prior to the addition of the us_state field in June 2023.
+  # For newer accounts, use the us_state field instead.
```

Comment:
Just curious, why not update `#us_state` of users created before June 2023 and use it without any conditions?

Reply:
Or we can do it for all the users created before June 2023 at once with migration or a few (depending on the data amount).
cc @Hamms

Reply:
I do thing a Backfill would be better here. How would a data backfill/migration lock older users?

If we backfill the data, we can keep the logic the same for all users, otherwise, this could be just the beginning of splitting code to treat different users. 
In 5 years we will be like "why this code?" and the answer will be something like "legacy reasons".

Reply:
> How would a data backfill/migration lock older users?

It's a bit nuanced because of how CPA is divided into two phases. This year, we need to ensure the lockout applies to new users. By requiring a state to be entered upon account creation, we are able to lock out 100% of new CO users. For existing accounts, we're OK until July 2024, except that if they add a new linked SSO account (e.g. Google) that counts as collecting new data that falls under the CPA umbrella.

If we backfill without changing the current lockout logic, it'll lock out existing accounts that we want to continue functioning normally except for being able to link new SSO accounts.

I think we can still accomplish this with a backfill as long as we also modify the existing lockout logic to include account creation date.